design/resource-validation.md

```diff
@@ -80,36 +91,9 @@ Design a validation mechanism for Gateway API resources.
 
 ## Design
 
-We will introduce two validation methods to be run by NGF control plane:
-
-1. Re-run of the Gateway API webhook validation
-2. NGF-specific field validation
-
-### Re-run of Webhook Validation
```

Comment:
I think keeping this section in the documentation makes sense since the code is still performing these additional checks, and we don't (yet) have plans to remove them.

Reply:
My thoughts were to simply mention in the NGF-validation section that we do validation on anything not covered in the CEL. That way we don't need this whole section.

Or at least simplify this section. We're really only doing one check that CEL doesn't do, right? Though it could expand in the future, but who knows.

Reply:
> We're really only doing one check that CEL doesn't do, right?

Sort of. There's redundancy right now because we support k8s 1.23 and 1.24. If the user is running 1.25+, CEL covers everything except one case that is caught by the webhook validation in our code. But regardless, we still run all the webhook checks as a safeguard.

Reply:
https://github.com/nginxinc/nginx-gateway-fabric/blob/f547884c6596f8d915e09ff6c9af8a4b1253b63d/internal/mode/static/state/change_processor_test.go#L1781 -- we rely on the web hook code so that we can make assumptions about the contents of the resources

Reply:
@sjberman 

> My thoughts were to simply mention in the NGF-validation section that we do validation on anything not covered in the CEL. That way we don't need this whole section.

> Or at least simplify this section. We're really only doing one check that CEL doesn't do, right? Though it could expand in the future, but who knows.

In addition to validation not done by CRDs,  NGF validation  assumes that CRD validation might not work properly so for values that end up in NGINX config, it sanitizes them. As a result, this prevents things like malicious config injection. https://github.com/nginxinc/nginx-gateway-fabric/blob/f547884c6596f8d915e09ff6c9af8a4b1253b63d/internal/mode/static/state/validation/validator.go#L3

Reply:
So would it be better to leave the re-running of the webhook on our end in? Thus, things would look like this:

Kubernetes 1.25+
1. OpenAPI validation
2. CEL validation
3. Webhook validation by NGF
4. Validation by NGF

Kubernetes 1.23 and 1.24
1. OpenAPI validation
2. Webhook validation by the Gateway API webhook?
3. Webhook validation by NGF
4. Validation by NGF